cpp/tests/io/orc_test.cpp

```diff
@@ -1000,47 +1010,77 @@ TEST_F(OrcStatisticsTest, Basic)
 
   auto const stats = cudf::io::read_parsed_orc_statistics(cudf::io::source_info{filepath});
 
-  auto const expected_column_names =
-    std::vector<std::string>{"", "_col0", "_col1", "_col2", "_col3", "_col4"};
+  auto expected_column_names = std::vector<std::string>{""};
+  std::generate_n(std::back_inserter(expected_column_names), expected.num_columns(), []() mutable {
+    static int starting_index = 0;
+    return "_col" + std::to_string(starting_index++);
+  });
```

Comment:
nit: 
is static required here?
how about?
```suggestion
  std::generate_n(std::back_inserter(expected_column_names), expected.num_columns(), [starting_index = 0]() mutable {
    return "_col" + std::to_string(starting_index++);
  });
```